internal/pkg/util/fs/helper.go

```diff
@@ -14,9 +14,36 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/sylabs/singularity/internal/pkg/sylog"
 	"golang.org/x/sys/unix"
 )
 
+// EnsureFilePrivate takes a file path, and 1. Creates it with
+// the specified permission, or 2. ensures a file path is the
+// specified permission.
+func EnsureFilePermission(fn string, mode os.FileMode) error {
+	fs, err := os.OpenFile(fn, os.O_RDWR|os.O_CREATE, mode)
+	if err != nil {
+		return err
+	}
+	defer fs.Close()
+
+	// check and fix permissions
+	fsinfo, err := fs.Stat()
+	if err != nil {
+		return err
+	}
+
+	if currentMode := fsinfo.Mode(); currentMode != mode {
+		sylog.Warningf("File mode (%o) on %s needs to be %o, fixing that...", currentMode, fn, mode)
```

Comment:
This message is a bit generic - I think if we change the perms we should give at least a hint of why they need to be `mode` for a specific file - which needs to be handled elsewhere.

Reply:
I added debug message where this is called, should it more verbose?

Reply:
Okay - it's not ideal as generally people we only see this Warning (they won't see debug messages), but we'll live with that.